jobserver/views/job_requests.py

```diff
@@ -109,16 +109,19 @@ def get_queryset(self):
                 # it then we can look for a job number
                 qs = qs.filter(qwargs | Q(jobs__pk=q))
 
+        def available_values(Model, field):
+            return {str(f) for f in Model.objects.values_list(field, flat=True)}
+
         backend = self.request.GET.get("backend")
-        if backend:
+        if backend and backend in available_values(Backend, "pk"):
```

Comment:
This approach feels a bit weird to me. If you supply an invalid value I'd expect the response to be a 400, rather than just for the value to be ignored. In fact, if you supply a syntactically valid value but one that doesn't exist in the db it will also be ignored, which seems even more problematic to me.

Could we define a function like `assert_is_int(value)` which raises an appropriate HTTP error if that value isn't an int?